website/blog/2024-09-24-run-ferretdb-cybertec-postgres-kubernetes.md

```diff
@@ -0,0 +1,350 @@
+```
+
+Output should look like this:
+
+```text
+{
+  "password": "naS0UMX4ajDUtFJZ2Zntwxscn5tnBnLsrDolSXqKOcxvaYkjAdjWRCRQhybbyORN",
+  "username": "postgres"
+}
+```
+
+The Postgres instance is ready for use as a backend for FerretDB.
+
+## Create and deploy FerretDB pods and service
+
+With the Postgres instance running, you need to create the FerretDB instance.
+The following `yaml` file sets up a FerretDB deployment and connects to the Postgres database using the `FERRETDB_POSTGRESQL_URL`.
+
+The host name for the Postgres instance is `cluster-1`.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ferretdb
+  namespace: cpo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ferretdb
+  template:
+    metadata:
+      labels:
+        app: ferretdb
+    spec:
+      containers:
+        - name: ferretdb
+          image: ghcr.io/ferretdb/ferretdb:latest
+          ports:
+            - containerPort: 27017
+          env:
+            - name: FERRETDB_POSTGRESQL_URL
+              value: postgres://postgres@cluster-1:5432/postgres
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ferretdb-service
+  namespace: cpo
+spec:
+  selector:
+    app: ferretdb
+  ports:
+    - name: mongo
+      protocol: TCP
+      port: 27017
+      targetPort: 27017
+```
+
+Apply the deployment by running:
+
+```sh
+kubectl apply -f ferretdb-deployment.yaml
+```
+
+This will create a FerretDB pod and service in the `cpo` namespace.
+
+Run `kubectl get pods -n cpo` to see that the pod is running.
+
+The output should look like this:
+
+```text
+NAME                                 READY   STATUS    RESTARTS   AGE
+cluster-1-0                          1/1     Running   0          17m
+ferretdb-8c5468db-mmqtr              1/1     Running   0          38s
+postgres-operator-78d4fdc97b-mp49x   1/1     Running   0          18m
+```
+
+You can also view the service by running `kubectl get svc -n cpo`.
+
+The output should look like this:
+
+```text
+% kubectl get svc -n cpo
+NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)     AGE
+cluster-1               ClusterIP   10.105.156.194   <none>        5432/TCP    36m
+cluster-1-clusterpods   ClusterIP   None             <none>        <none>      36m
+cluster-1-repl          ClusterIP   10.101.77.119    <none>        5432/TCP    36m
+ferretdb-service        ClusterIP   10.102.252.125   <none>        27017/TCP   19m
+```
+
+## Access FerretDB instance via `mongosh`
+
+You can connect to the FerretDB instance using `mongosh`.
+
+Start by creating a temp `mongosh` pod:
+
+```sh
+kubectl run -it --rm --image=mongo:latest mongo-client -- bash
+```
+
+Connect to the FerretDB instance using the Postgres credentials generated earlier:
+
+```sh
+mongosh "mongodb://postgres:<password>@host:27017/postgres?authMechanism=PLAIN"
```

Comment:
```suggestion
mongosh "mongodb://postgres:<password>@<host>:27017/postgres?authMechanism=PLAIN"
```

Just so that it's consistent with other variables.